rln-js/index.html

```diff
@@ -155,27 +161,64 @@
         updateFields()
     }
 
-    const ContentTopic = "/toy-chat/2/luzhou/proto";
-    // const ContentTopic = "/toy-chat/2/huilong/proto"
-    const decoder = new DecoderV0(ContentTopic);
-
     let messages = [];
 
-    const updateMessages = (msgs, div) => {
-        div.innerHTML = "<ul>"
-        messages.forEach(msg => div.innerHTML += "<li>" + msg + "</li>")
-        div.innerHTML += "</ul>"
+    const updateMessages = () => {
+        messagesDiv.innerHTML = "<ul>"
+        messages.forEach(msg => {
+            messagesDiv.innerHTML += `<li>${msg.msg} - [epoch: ${msg.epoch}, proof: ${msg.proofState} ]</li>`
+
+            if (msg.proofState === "verifying...") {
+                console.log("Verifying proof")
+                const res = msg.verify()
+                console.log("proof verified!", res)
+                if (res === undefined) {
+                    msg.proofState = "no proof attached"
+                } else if (res) {
+                    msg.proofState = "verified."
+                } else {
+                    msg.proofState = "invalid!"
+                }
+                updateMessages()
+            }
+        })
+        messagesDiv.innerHTML += "</ul>"
     }
 
     const callback = (wakuMessage) => {
         const {timestamp, nick, text} = ProtoChatMessage.decode(wakuMessage.payload)
         const time = new Date();
         time.setTime(Number(timestamp));
 
-        messages.push(`(${nick}) <strong>${utils.bytesToUtf8(text)}</strong> <i>[${time.toString()}]</i>`);
-        updateMessages(messages, messagesDiv)
+        console.log("msg:", wakuMessage)
+
+        let proofState, verify;
+        if (typeof wakuMessage.verify === "undefined") {
+            proofState = "no proof attached";
+        } else {
+            verify = wakuMessage.verify;
+            proofState = "verifying...";
+        }
+
+        console.log("verify", wakuMessage.verify)
+
+        console.log("proof", wakuMessage.rateLimitProof)
```

Comment:
This shows

![image](https://user-images.githubusercontent.com/110212804/193205099-287afcf7-7072-4270-a9b5-e58f21e2853e.png)

as you can see the `rlnIdentifier` is an empty array. the `proof` is 256 bytes